stock_move_auto_assign/models/stock_move.py

```diff
@@ -54,10 +54,12 @@ def _prepare_auto_assign(self, location_field):
         for product_id, location_ids in product_locs.items():
             if not location_ids:
                 continue
-            self._enqueue_auto_assign(
+            job = self._enqueue_auto_assign(
                 self.env["product.product"].browse(product_id),
                 self.env["stock.location"].browse(location_ids),
-            ).delay()
+            )
+            if job:
+                job.delay()
 
     def _enqueue_auto_assign(self, product, locations, **job_options):
```

Comment:
let's make clear self (move) can't be use
```suggestion
    @api.model
    def _enqueue_auto_assign(self, product, locations, **job_options):
```